src/dataStructures/binarySearchTree.js

```diff
@@ -0,0 +1,53 @@
+const TreeNode = require('./treeNode.js');
+
+module.exports = class BinarySearchTree {
+  constructor(args) {
+    this.head = null;
+    if (Array.isArray(args) && args.length !== 0) {
+      this.addNodesFromArray(args);
+    } else if (Number.isInteger(args)) {
+      this.addNode(args);
+    } else {
+      console.log('Sorry, please check your input params.');
```

Comment:
Is there anyway we can have it such that a BinarySearchTree can be initialized without a parameter?

Also - it should be able to hold any form of data (not just integers).